src/TypeSystem.h

```diff
@@ -388,6 +388,7 @@ class TypeEnvironment {
 /**
  * Returns full type qualifier for a given type
  */
+// TODO: maybe use this instead of names for types later on
```

Comment:
Use the usual `// TODO (name/issue): stuff` format so these are easier to find and remove (/do) later

Reply:
I'm using the `// TODO (name/issue): stuff` format as a mark for TODOs that should be completed later, while unmarked TODOs introduced all have to be thought about before the final (next) PR of the stage. Happy to mark them as temporary TODOs, though.